addons/point_of_sale/models/pos_payment.py

```diff
@@ -65,7 +65,11 @@ def export_for_ui(self):
 
     def _create_payment_moves(self):
         result = self.env['account.move']
-        for payment in self:
+        cash_payment = self.filtered(lambda p: p.payment_method_id.type == 'cash')
+        change_payment = self.filtered(lambda p: p.is_change and p.payment_method_id.type == 'cash')
+        if cash_payment and change_payment:
+            cash_payment[0].amount += change_payment.amount
+        for payment in self.filtered(lambda p: not p.is_change):
```

Comment:
What if there is no cash payment methods but there is a change ?

Reply:
I changed the logic so the change amount will be added to the first cash payment or, if none, the first non-change payment, what do you think?